x-pack/plugins/security_solution/public/common/components/endpoint/endpoint_agent_status/endpoint_agent_status.tsx

```diff
@@ -0,0 +1,343 @@
+   * If undefined, then this component will call the API to retrieve that list of pending actions.
+   * NOTE: if this prop is defined, it will invalidate `autoRefresh` prop.
+   */
+  pendingActions?: EndpointPendingActions['pending_actions'];
+  'data-test-subj'?: string;
+}
+
+/**
+ * Displays the status of an Endpoint agent along with its Isolation state or the number of pending
+ * response actions against it.
+ *
+ * TIP: if you only have the Endpoint's `agent.id`, then consider using `EndpointAgentStatusById`,
+ * which will call the needed APIs to get the information necessary to display the status.
+ */
+export const EndpointAgentStatus = memo<EndpointAgentStatusProps>(
+  ({ endpointHostInfo, autoRefresh = true, pendingActions, 'data-test-subj': dataTestSubj }) => {
+    const getTestId = useTestIdGenerator(dataTestSubj);
+    const { data: endpointPendingActions } = useGetEndpointPendingActionsSummary(
+      [endpointHostInfo.metadata.agent.id],
+      {
+        refetchInterval: autoRefresh ? DEFAULT_POLL_INTERVAL : false,
+        enabled: !pendingActions,
+      }
+    );
+
+    const [hasPendingActions, hostPendingActions] = useMemo<
+      [boolean, EndpointPendingActions['pending_actions']]
+    >(() => {
+      if (!endpointPendingActions && !pendingActions) {
+        return [false, {}];
+      }
+
+      const pending = pendingActions
+        ? pendingActions
+        : endpointPendingActions?.data[0].pending_actions ?? {};
+
+      return [Object.keys(pending).length > 0, pending];
+    }, [endpointPendingActions, pendingActions]);
+
+    const status = endpointHostInfo.host_status;
+    const isIsolated = Boolean(endpointHostInfo.metadata.Endpoint.state?.isolation);
+
+    return (
+      <EuiFlexGroupStyled
+        gutterSize="none"
+        responsive={false}
+        className="eui-textTruncate"
+        data-test-subj={dataTestSubj}
+      >
+        <EuiFlexItem grow={false}>
+          <EuiBadge
+            color={status != null ? HOST_STATUS_TO_BADGE_COLOR[status] : 'warning'}
+            data-test-subj={getTestId('agentStatus')}
+            className="eui-textTruncate"
+          >
+            {getAgentStatusText(status)}
+          </EuiBadge>
+        </EuiFlexItem>
+        {(isIsolated || hasPendingActions) && (
+          <EuiFlexItem grow={false} className="eui-textTruncate isolation-status">
+            <EndpointHostResponseActionsStatus
+              data-test-subj={getTestId('actionStatuses')}
+              isIsolated={isIsolated}
+              pendingActions={hostPendingActions}
+            />
+          </EuiFlexItem>
+        )}
+      </EuiFlexGroupStyled>
+    );
+  }
+);
+EndpointAgentStatus.displayName = 'EndpointAgentStatus';
+
+export interface EndpointAgentStatusByIdProps {
+  endpointAgentId: string;
+  /**
+   * If set to `true` (Default), then the endpoint status and isolation/action counts will
+   * be kept up to date by querying the API periodically
+   */
+  autoFresh?: boolean;
```

Comment:
Maybe you missed this one. We can alsochange in a new PR.

Reply:
FYI: Done in https://github.com/elastic/kibana/pull/155455